drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -92,6 +97,50 @@ static int udc_mcux_control(const struct device *dev, usb_device_control_type_t
 	return 0;
 }
 
+/* Helper function to keep track of the activity state for the udc */
+static void udc_mcux_change_state(const struct device *dev, bool enabled, bool vbus_present)
+{
+	struct udc_mcux_data *data = udc_get_private(dev);
+
+	if (data->enabled == enabled && data->vbus_present == vbus_present) {
+		return;
+	}
+
+	if (vbus_present != data->vbus_present && enabled) {
+		udc_submit_event(data->dev,
+				 vbus_present ? UDC_EVT_VBUS_READY : UDC_EVT_VBUS_REMOVED, 0);
+	}
+	if (enabled && vbus_present) {
+		data->enabled = true;
+		data->vbus_present = true;
+
+		/*
+		 * Block PM when usb is active.
+		 */
+#if defined(CONFIG_PM_POLICY_DEVICE_CONSTRAINTS)
+		pm_policy_device_power_lock_get(dev);
+#endif
```

Comment:
No #if/#endif, that needs to be fixed in pm subsytem.

Reply:
@jfischer-no, do you want this author to change the underlying pm subsystem here with this PR?

Reply:
Fixed with PR https://github.com/zephyrproject-rtos/zephyr/pull/90938 please review :)